server/mediaserver.go

```diff
@@ -1037,6 +1038,97 @@ func getPlaylistsFromStore(ctx context.Context, sess drivers.OSSession, manifest
 	return filesMap, jsonFiles, latestPlaylistTime, nil
 }
 
+func (s *LivepeerServer) streamMP4(w http.ResponseWriter, r *http.Request, jpl *core.JsonPlaylist, manifestID, track string) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	contentType, _ := common.TypeByExtension(".mp4")
+	w.Header().Set("Content-Type", contentType)
+	w.Header().Set("Connection", "keep-alive")
+	var sourceBytesSent, resultBytesSent int64
+
+	tc := ffmpeg.NewTranscoder()
+	or, ow, err := os.Pipe()
+	if err != nil {
+		glog.Errorf("manifestID=%s error creating pipe err=%v", manifestID, err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	done := make(chan struct{})
+	go func() {
+		resultBytesSent, _ = io.Copy(w, or)
+		or.Close()
+		done <- struct{}{}
+	}()
+	stop := func() {
```

Comment:
Can we just defer this?

Reply:
changed to defer in e3a1a3f3925366193b21ad0cfa51d7f3e4cb626c